src/Confluent.Kafka/Consumer.cs

```diff
@@ -551,6 +506,43 @@ private static byte[] ValueAsByteArray(rd_kafka_message msg)
         }
 
 
+        /// <summary>
+        ///     Poll for new messages / events. Blocks until a consume result
+        ///     is available or the operation has been cancelled.
+        /// </summary>
+        /// <param name="cancellationToken">
+        ///     A cancellation token that can be used to cancel this operation.
+        /// </param>
+        /// <returns>
+        ///     The consume result.
+        /// </returns>
+        /// <remarks>
+        ///     OnPartitionsAssigned/Revoked, OnOffsetsCommitted and
+        ///     OnPartitionEOF events may be invoked as a side-effect of
+        ///     calling this method (on the same thread).
+        /// </remarks>
+        public ConsumeResult Consume(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var result = Consume(100, Deserializers.ByteArray, Deserializers.ByteArray);
```

Comment:
The value 100 is hardcoded, can this be parameterized, please?

Reply:
good idea @neeraj2k6, will do. however I'm going to merge this PR up as is and do this as a separate change.